### DISCUSSION TITLE
How to configure Deno

### DISCUSSION THREAD

**Discussion by JasterV** [LazyVim/LazyVim#674] [Q&A] [answered]
I'm trying to get this working 
```
@folke Can you explain why does the neoconf plugin need to be removed and what consequences might that have?
I'm trying to get this working

  {
    "neovim/nvim-lspconfig",
    ---@class PluginLspOpts
    opts = {
      ---@type lspconfig.options
      servers = {
        denols = {
          root_dir = require("lspconfig.util").root_pattern("deno.json", "deno.jsonc"),
        },
        tsserver = {
          root_dir = require("lspconfig.util").root_pattern("package.json"),
          single_file_support = false,
        },
      },
    },
  },
```

As explained [here](https://deno.com/manual@v1.32.5/getting_started/setup_your_environment#neovim-06-using-the-built-in-language-server) this is required to have `denols` working properly.

I saw on another discussion that @folke mentions that a plugin needs to be disabled to have the `root_dir` property working, how does that work? Can I get more detail on how the `root_dir` works and why does a plugin need to be disabled?

**Comment by folke** [comment, marked as answer]
You no longer need to add those root_dir options. In LazyVim it will all work out of the box. You also no longer need to disable neoconf

**Reply by leighmcculloch** [reply]
I'm using LazyVim out of the box and not getting denotls to trigger, vtsls still seems to be the LSP running for .ts files inside a repo with a deno.jsonc. Is there something we need to do to enable denotls?

**Reply by leighmcculloch** [reply]
I think the following is what's required. Is that the correct way to configure it?

```lua
return {
  "neovim/nvim-lspconfig",
  opts = {
    servers = {
      denols = {
        enabled = true,
      },
    },
  },
}
```
